pkg/asset/agent/gencrypto/authconfig.go

```diff
@@ -54,12 +77,28 @@ func (a *AuthConfig) Generate(_ context.Context, dependencies asset.Parents) err
 
 	a.PublicKey = encodedPubKeyPEM
 
-	token, err := localJWTForKey(infraEnvID.ID, privateKey)
+	newAgentAuthToken, err := LocalJWTForKey(infraEnvID.ID, privateKey, agentWorkflow.Workflow, expiration, expirationUnit)
```

Comment:
For the addnodes workflow, I would prefer to only generate a new token if an existing token on the cluster cannot be used. Currently it replaces the one in the asset if the cluster contains a valid token.

Reply:
I will revisit this option and address an issue with the current implementation. The code currently generates a public key and signs a new token with a private key associated with that public key. However, if the token in the cluster remains valid during the first 24 hours out of 48 hours, the asset store token is updated with the value from the cluster. This is problematic because the public key stored in the asset store does not match the one in the cluster. Consequently, when the public key is passed to the assisted service for verification, the token verification fails as it attempts to verify a token signed with a different private key not associated with the available public key

Reply:
Thanks for the clarification @pawanpinjarkar. After the discussion, I also agree that in case of refresh _both_ the token _and_ the pub key must be refreshed (and thus stored in the cluster secret)